### PR TITLE
Revert "Add `"lamb"` to `str2optimizer32bit`"

### PR DESCRIPTION
**Pull request by younesbelkada**
Reverts TimDettmers/bitsandbytes#1222 & Fixes https://github.com/TimDettmers/bitsandbytes/issues/1118

cc @Titus-von-Koeller @matthewdouglas 

This seems to cause undefined symbol issues for users that have bnb installed from source. It seems we forgot to add proper C-methods for each newly added optimizer

### CODE REVIEW

**Comment by BenjaminBossan** [comment]
Also ping @EtienneDosSantos

File: bitsandbytes/functional.py [hunk 1]
```diff
@@ -27,128 +27,63 @@ def prod(iterable):
...
-            lib.cmomentum32bit_grad_fp32,
-            lib.cmomentum32bit_grad_fp16,
-        ),
-        "rmsprop": (
-            lib.crmsprop32bit_grad_fp32,
-            lib.crmsprop32bit_grad_fp16,
+        "adagrad": (
+            lib.cadagrad32bit_grad_32,
+            lib.cadagrad32bit_grad_16,
         ),
```
Comment (younesbelkada) [inline bitsandbytes/functional.py:51]:
```suggestion
        ),
        "lamb": (
            lib.cadam32bit_grad_fp32,
            lib.cadam32bit_grad_fp16,
        ),
```

Comment (younesbelkada) [reply bitsandbytes/functional.py:51]: This is the commit: https://github.com/TimDettmers/bitsandbytes/pull/1222/commits/a9a1c44abaf671f06794f6664fe3ca47bcade5e5 that did not cause the issues

**Comment by younesbelkada** [comment]
I can confirm with this PR import of bnb works fine and fixes at the same time the issue mentioned in #1222

**Comment by Titus-von-Koeller** [comment]
Thanks Younes ❤️

**Comment by EtienneDosSantos** [comment]
Thx for fixing!

**Pull request merged**